oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/OAuth2AccessToken.java

```diff
@@ -128,4 +150,19 @@ public int hashCode() {
 			return this.getValue().hashCode();
 		}
 	}
+
+	@Override
+	public Instant getIssuedAt() {
+		return getClaimAsInstant("iat");
```

Comment:
The `iat` claim is defined by the OpenID Connect related specifications as well as `Jwt` spec. However, this claim is likely not available for standard OAuth 2.0 providers like GitHub and Facebook. But it doesn't mean they don't provide their own attribute value for `issueAt` but using a different attribute name.

Reply:
couldn't it be solved with more polymorphisme?